spec/lib/blacklight/document/cache_key_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'spec_helper'
+
+describe Blacklight::Document::CacheKey do
+  let(:attributes) { {} }
+  let(:subject) { SolrDocument.new(attributes) }
+  it 'SolrDocument includes the module' do
+    expect(subject.class).to include(Blacklight::Document::CacheKey)
+  end
+
+  describe 'new record' do
+    before do
+      allow(subject).to receive_messages(new_record?: true)
+    end
+    it 'provides an acceptable cache key' do
+      expect(subject.cache_key).to eq 'solr_documents/new'
+    end
+  end
+
+  describe 'with version' do
+    let(:attributes) { { id: '12345', _version_: '1497353774427013120' } }
+    it 'provides a cache key with the id and version' do
+      expect(subject.cache_key).to eq 'solr_documents/12345-1497353774427013120'
+    end
+    describe 'as array' do
+      let(:attributes) { { id: '12345', _version_: ['1234', '4321'] } }
+      it 'provides a cache key with the id and joined version array' do
+        expect(subject.cache_key).to eq 'solr_documents/12345-12344321'
+      end
+    end
+  end
+
+  describe 'without version' do
+    let(:attributes) { { id: '12345' } }
+    it 'provides a cache key with just the id' do
+      expect(subject.cache_key).to eq 'solr_documents/12345'
+    end
+  end
+
+  describe '#cache_version_key' do
+    let(:attributes) { { id: '12345', another_version_field: '1497353774427013120' } }
```

Comment:
Line is too long. [86/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.